src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -64,7 +64,7 @@
     /**
      * A submission can have a result and therefore, results are persisted and removed with a submission.
      */
-    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY)
+    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE, orphanRemoval = true)
```

Comment:
This change was done in https://github.com/ls1intum/Artemis/pull/695/files#diff-29ba23544d01d5a0e5f8f40af6d2ddc5R67 to improve the programming exercise submission workflow.

What is the reason that this needed to be changed? Is this a new bug that we introduced with the other PR?